tests/Test.ADAL.NET.Integration/SovereignCloudsTests.cs

```diff
@@ -1,75 +1,162 @@
+using Test.ADAL.NET.Common;
+using Test.ADAL.NET.Common.Mocks;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.OAuth2;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System;
 
-namespace Test.ADAL.NET.Unit
+namespace Test.ADAL.NET.Integration
 {
     [TestClass]
     public class SovereignCloudsTests
     {
-        private PlatformParameters platformParameters;
+        private PlatformParameters _platformParameters;
+
+        private const string SovereignAuthorityHost = "login.microsoftonline.de";
+
+        private readonly string _sovereignTenantSpecificAuthority = String.Format("https://{0}/{1}/", SovereignAuthorityHost, TestConstants.SomeTenantId);
 
         [TestInitialize]
         public void Initialize()
         {
-            HttpMessageHandlerFactory.ClearMockHandlers();
-            platformParameters = new PlatformParameters(PromptBehavior.Auto);
+            HttpMessageHandlerFactory.InitializeMockProvider();
+            _platformParameters = new PlatformParameters(PromptBehavior.Auto);
+            InstanceDiscovery.InstanceCache.Clear();
         }
 
-/*        [TestMethod]
+        [TestMethod]
         [Description("Sovereign user use world wide authority")]
         public async Task SovereignUserWorldWideAuthorityIntegrationTest()
         {
-            const string sovereignAuthorityHost = "login.some-sovereign-cloud";
-
-            var sovereignTenantSpesificAuthority = $"https://{sovereignAuthorityHost}/{TestConstants.SomeTenantId}/";
-
             // creating AuthenticationContext with common Authority
             var authenticationContext =
                 new AuthenticationContext(TestConstants.DefaultAuthorityCommonTenant, false, new TokenCache());
 
             // mock value for authentication returnedUriInput, with cloud_instance_name claim
             var authReturnedUriInputMock = TestConstants.DefaultRedirectUri + "?code=some-code" + "&" +
-                                           TokenResponseClaim.CloudInstanceHost + "=" + sovereignAuthorityHost;
+                                           TokenResponseClaim.CloudInstanceHost + "=" + SovereignAuthorityHost;
 
             MockHelpers.ConfigureMockWebUI(
                 new AuthorizationResult(AuthorizationStatus.Success, authReturnedUriInputMock),
                 // validate that authorizationUri passed to WebUi contains instance_aware query parameter
-                new Dictionary<string, string> {{"instance_aware", "true"}});
+                new Dictionary<string, string> { { "instance_aware", "true" } });
+
+            HttpMessageHandlerFactory.AddMockHandler(MockHelpers.CreateInstanceDiscoveryMockHandler(TestConstants.GetDiscoveryEndpoint(TestConstants.DefaultAuthorityCommonTenant)));
 
             HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
```

Comment:
why the url not being enforced here? Travis made the change already. it should force you to pass in url

Reply:
updated this.